controllers/neutronsriovagent_controller.go

```diff
@@ -1,163 +1,116 @@
-func (r *ReconcileNeutronSriovAgent) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+// Reconcile reconcile keystone API requests
+// +kubebuilder:rbac:groups=neutron.openstack.org,resources=neutronsriovagents,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=neutron.openstack.org,resources=neutronsriovagents/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;create;update;delete;
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;create;update;delete;
+func (r *NeutronSriovAgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	reqLogger := r.Log.WithValues("neutronsriovagent", req.NamespacedName)
 	reqLogger.Info("Reconciling NeutronSriovAgent")
 
 	// Fetch the NeutronSriovAgent instance
-	instance := &neutronv1.NeutronSriovAgent{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	instance := &neutronv1beta1.NeutronSriovAgent{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			return reconcile.Result{}, nil
+			return ctrl.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		return ctrl.Result{}, err
 	}
 
 	commonConfigMap := &corev1.ConfigMap{}
 
 	reqLogger.Info("Creating host entries from config map:", "configMap: ", CommonConfigMAP)
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: CommonConfigMAP, Namespace: instance.Namespace}, commonConfigMap)
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: CommonConfigMAP, Namespace: instance.Namespace}, commonConfigMap)
 	if err != nil && errors.IsNotFound(err) {
 		reqLogger.Error(err, "common-config ConfigMap not found!", "Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
-		return reconcile.Result{}, err
+		return ctrl.Result{}, err
 	}
 
-	if err := controllerutil.SetControllerReference(instance, commonConfigMap, r.scheme); err != nil {
-		return reconcile.Result{}, err
+	if err := controllerutil.SetControllerReference(instance, commonConfigMap, r.Scheme); err != nil {
+		return ctrl.Result{}, err
 	}
 
 	// Create additional host entries added to the /etc/hosts file of the containers
 	ospHostAliases, err = util.CreateOspHostsEntries(commonConfigMap)
```

Comment:
we no longer need to add the hosts entries as we now have dns service from OCP. maybe in future we want to add it again where external services are required.